iothub/device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/mqtt/MqttMessaging.java

```diff
@@ -40,18 +50,31 @@ public MqttMessaging(
             throw new IllegalArgumentException("Device id cannot be null or empty");
         }
 
-        if (moduleId == null || moduleId.isEmpty())
-        {
-            this.publishTopic = "devices/" + deviceId + "/messages/events/";
-            this.eventsSubscribeTopic = "devices/" + deviceId + "/messages/devicebound/#";
-            this.inputsSubscribeTopic = null;
-        }
-        else
-        {
-            this.publishTopic = "devices/" + deviceId + "/modules/" + moduleId +"/messages/events/";
-            this.eventsSubscribeTopic = "devices/" + deviceId + "/modules/" + moduleId + "/messages/devicebound/#";
-            this.inputsSubscribeTopic = "devices/" + deviceId + "/modules/" + moduleId +"/inputs/#";
-        }
+        String clientId = (moduleId == null || moduleId.isEmpty()) ?
+                deviceId :
+                deviceId + BACKSLASH + MODULES + BACKSLASH + moduleId;
+
+        // Publish to "$iothub/devices/{clientId}/messages/events/"
+        this.publishTopic = DEVICES +
+                BACKSLASH +
+                clientId +
+                BACKSLASH +
+                EVENTS +
+                BACKSLASH;
+
+        // Subscribe to "$iothub/devices/{clientId}/messages/devicebound/#"
```

Comment:
Ideally we would have a constant declared for this string (but for the client Id). That way, we can just do 
```String.format(DEVICEBOUND_TOPIC, clientId)```
instead of all this concatenation below